tests/src/Mod/Part/App/FeatureExtrusion.cpp

```diff
@@ -298,3 +299,31 @@ TEST_F(FeatureExtrusionTest, testExecuteFaceMaker)
     EXPECT_FLOAT_EQ(volume, len * wid * ext1);
     EXPECT_TRUE(PartTestHelpers::boxesMatch(bb, Base::BoundBox3d(0, 0, 0, len, wid, ext1)));
 }
+
+TEST_F(FeatureExtrusionTest, testFaceWithHoles)
+{
+    // Arrange
+    float radius = 0.75;
+    auto [face1, wire1, wire2] = PartTestHelpers::CreateFaceWithRoundHole(len, wid, radius);
+    // face1 is the sum of the outside (wire1) and the internal hole (wire2).
+    Part::TopoShape newFace = Part::TopoShape(face1).makeElementFace(nullptr);
+    // newFace cleans that up and is the outside minus the internal hole.
+    auto face2 = newFace.getShape();
+
+    auto partFeature = dynamic_cast<Part::Feature*>(_doc->addObject("Part::Feature"));
+    partFeature->Shape.setValue(face2);
+    _extrusion->Base.setValue(_doc->getObjects().back());
+    _extrusion->FaceMakerClass.setValue("Part::FaceMakerCheese");
+    // Act
+    _extrusion->execute();
+    Part::TopoShape ts = _extrusion->Shape.getValue();
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    Base::BoundBox3d bb = ts.getBoundBox();
+    // Assert
+    EXPECT_FLOAT_EQ(volume,
+                    len * wid * ext1
+                        - radius * radius * M_PI * ext1);  // Why isn't this be a subtraction?
```

Comment:
Can you explain the question here?

Reply:
Ha!  Yeah.  It was a plus, which didn't make sense, until I found the issue and fixed it.  Now it is a subtraction like it should be.  But I missed the comment I left there, which is about to get deleted.  The issue is that the BRepBuilderAPI_MakeFace(face1, wire2).Face(); in CreateFaceWithRoundHole gives a counter-intuitive result unless you do the makeElementFace.